packages/analytics-adapter/src/awaiter.ts

```diff
@@ -0,0 +1,80 @@
+
+export abstract class Awaiter {
```

Comment:
Just a thought: it seems to me that `waitForAnalytics()` and `waitForEvolv()` could be generalized into something like a loop that wires up helpers like [waitFor](https://github.com/evolv-ai/asset-manager/blob/1c5044f8eb48136e1d674b3a01ee444f6ff71bef/src/helpers.js#L12-L42).

```typescript
interface Prerequisite {
    predicate: () => boolean;
    callback: () => void;
    timeout?: number;
}

class Awaiter {
	constructor(prerequisites: Prerequisites[]) {
		for (const { predicate, callback, timeout } of prerequisites) {
        		waitFor(predicate, callback, timeout);
		}
	}
}
```